src/custom.rs

```diff
@@ -78,9 +78,9 @@ macro_rules! register_custom_getrandom {
     ($path:path) => {
         // We use an extern "C" function to get the guarantees of a stable ABI.
         #[no_mangle]
-        extern "C" fn __getrandom_custom(dest: *mut u8, len: usize) -> u32 {
+        extern "C" fn __getrandom_custom(dst: *mut u8, len: usize) -> u32 {
             let f: fn(&mut [u8]) -> Result<(), $crate::Error> = $path;
-            let slice = unsafe { ::core::slice::from_raw_parts_mut(dest, len) };
+            let slice = unsafe { ::core::slice::from_raw_parts_mut(dst, len) };
```

Comment:
I think this is incorrect and fixing it would break backward compatibility.

`getrandom_raw` can be called with a pointer to uninitialized memory. So, we cannot legally construct a `&mut [u8]` slice from that pointer since we're only allowed to construct such a slice if the pointer points to initialized memory.

The function `$path` has a parameter of type `&mut [u8]` but to support the `_raw` API and/or the `_uninit` API it must take `&mut [MaybeUninit<u8>]` or a raw pointer and length instead.